ambari/datadog_checks/ambari/ambari.py

```diff
@@ -38,40 +36,30 @@ def check(self, instance):
 
         clusters = self.get_clusters(clusters_endpoint)
         self.get_hosts_metrics(clusters, server, port, tags)
-        
 
-        
     def get_clusters(self, url):
         cluster_list = []
-        try:
-            r = self.http.get(url)
-            r.raise_for_status()
-        except:
-            raise
+        r = self.http.get(url)
+        r.raise_for_status()
 
         clusters = r.json().get('items')
         for cluster in clusters:
             cluster_list.append(cluster.get('Clusters').get('cluster_name'))
         return cluster_list
 
     def get_hosts_metrics(self, clusters, server, port, tags):
-
         for cluster in clusters:
             hosts_endpoint = HOSTS_URL.format(ambari_server=server, ambari_port=port, cluster_name=cluster)
+            resp = self.http.get(hosts_endpoint)
+            resp.raise_for_status()
+            hosts_list = resp.json().get('items')
+            cluster_tag = "ambari_cluster:{}".format(cluster)
 
-            try:
-                resp = self.http.get(hosts_endpoint)
-                resp.raise_for_status()
-            except:
-                raise
-
-        hosts_list = resp.json().get('items')
```

Comment:
resp is define inside the for loop, this line was not inside the for loop so you will only be executing this for the last cluster. I've moved it inside the for loop so it runs for every cluster

Reply:
Good point.  I will need to reconsider this though as more of adding hosts to the list so it doesn't get reset each time the for loop runs.